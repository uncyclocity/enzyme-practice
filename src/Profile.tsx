export default function Profile({
  username,
  name,
}: {
  username: string;
  name: string;
}) {
  return (
    <div>
      <b>{username}</b>
      <span>({name})</span>
    </div>
  );
}
