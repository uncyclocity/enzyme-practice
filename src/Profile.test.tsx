import { mount } from "enzyme";
import Profile from "./Profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<Profile username="uncyclocity" name="백괴" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders username and name", () => {
    const wrapper = mount(<Profile username="uncyclocity" name="백괴" />);
    expect(wrapper.props().username).toBe("uncyclocity");
    expect(wrapper.props().name).toBe("백괴");
    const boldElement = wrapper.find("b");
    expect(boldElement.contains("uncyclocity")).toBe(true);
    const spanElement = wrapper.find("span");
    expect(spanElement.text()).toBe("(백괴)");
  });
});
