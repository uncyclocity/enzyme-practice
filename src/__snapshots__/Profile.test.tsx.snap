// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profile
  name="氚标创"
  username="uncyclocity"
>
  <div>
    <b>
      uncyclocity
    </b>
    <span>
      (
      氚标创
      )
    </span>
  </div>
</Profile>
`;
